src/Qir/Common/cmake/qir_cmake_include.cmake

```diff
@@ -65,4 +65,46 @@ macro(target_source_from_qir_obj target_name source_file)
     target_sources(${target_name} PUBLIC 
         "${CMAKE_CURRENT_BINARY_DIR}/${source_file}.obj"
     )
-endmacro()
\ No newline at end of file
+endmacro()
+
+#===============================================================================
+# compiling from IR
+macro(target_source_from_qir target_name source_file)
+    set(CLANG_ARGS "-c")
+    if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+        set(CLANG_ARGS
+        "${CLANG_ARGS}"
+        "-O0"
+        "-D_DEBUG"
+        )
+    endif()
```

Comment:
(Minor)
Among the debugging-related macros consider using the `NDEBUG` only. As I know, this is the only macro mentioned in the standard of C and C++. The other ones, like `_DEBUG`, `DEBUG`, are non-standard.

Reply:
Looking around at our codebase, it looks like we actually use `DEBUG` everywhere else (the rest of the QIR code use `set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -DDEBUG")`, see for example [here](https://github.com/microsoft/qsharp-runtime/blob/swernli/qir-build-native/src/Qir/Runtime/CMakeLists.txt#L12)). Given that, I'm going to update this location to use `DEBUG` instead of `_DEBUG` for consistency, and we can consider adopting to the alternate pattern later by updating all the locations together in a targeted PR.

Reply:
I also realized looking back at this file that I forgot to remove the old QIR compilation macro which is no longer used!

Reply:
FYI: I have opened #600 (Consider using NDEBUG instead of DEBUG, _DEBUG) to not forget about this.